skyportal/handlers/api/gcn.py

```diff
@@ -0,0 +1,202 @@
+        ).all()
+        if len(localization) == 0:
+            DBSession().merge(Localization(**skymap))
+            DBSession().commit()
+        localization = Localization.query.filter_by(
+            dateobs=dateobs, localization_name=skymap["localization_name"]
+        ).one()
+
+        localization = get_contour(localization)
+        DBSession().merge(localization)
+        DBSession().commit()
+
+        return self.success()
+
+
+default_prefs = {'maxNumGcnEvents': 10, 'sinceDaysAgo': 3650}
+
+
+class GcnEventViewsHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        user_prefs = getattr(self.current_user, 'preferences', None) or {}
+        top_events_prefs = user_prefs.get('topGcnEvents', {})
+        top_events_prefs = {**default_prefs, **top_events_prefs}
+
+        max_num_events = int(top_events_prefs['maxNumGcnEvents'])
+        since_days_ago = int(top_events_prefs['sinceDaysAgo'])
+
+        cutoff_day = datetime.datetime.now() - datetime.timedelta(days=since_days_ago)
+        q = GcnEvent.query
+
+        events = []
+        for event in q.all():
+            if len(events) >= max_num_events:
+                continue
+            dateobs = event.dateobs
+            if dateobs < cutoff_day:
+                continue
+            tags = [_ for _ in event.tags]
+            localizations = [_.localization_name for _ in event.localizations]
+            events.append(
+                {'localizations': localizations, 'dateobs': dateobs, 'tags': tags}
+            )
+
+        return self.success(data=events)
+
+
+class GcnEventHandler(BaseHandler):
+    @auth_or_token
+    def get(self, dateobs):
+        q = GcnEvent.query.filter_by(dateobs=dateobs)
+        event = q.first()
+        tags = [_ for _ in event.tags]
+        localizations = [_.localization_name for _ in event.localizations]
+        notices = [_.content for _ in event.gcn_notices]
+        data = [
+            {
+                'tags': tags,
+                'dateobs': event.dateobs,
+                'localizations': localizations,
+                'lightcurve': event.lightcurve,
+                'notices': notices,
+            }
+        ]
+        return self.success(data=data)
+
+
+class LocalizationHandler(BaseHandler):
+    @auth_or_token
+    def get(self, dateobs, localization_name):
+        q = Localization.query.filter_by(
+            dateobs=dateobs, localization_name=localization_name
+        )
+        localization = q.first()
+
+        data = {
+            'flat_2d': localization.flat_2d,
+            'contour': localization.contour,
+            'dateobs': localization.dateobs,
+            'localization_name': localization.localization_name,
+        }
+
+        return self.success(data=data)
+
+
+class ZTFFieldsHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        contours = []
+        import numpy as np
+        import copy
+        import gwemopt.utils
```

Comment:
Why are these imports in the method definition rather than at the top of the file?